legal-api/src/legal_api/resources/business/business_filings.py

```diff
@@ -392,19 +392,32 @@ def _is_before_epoch_filing(filing_json: str, business: Business):
             filing_json['filing']['header']['date']).replace(tzinfo=datetime.timezone.utc)
         return filing_date < epoch_filing[0].filing_date
 
+    @staticmethod
+    def _is_historical_colin_filing(filing_json: str):
+        if filing_json['filing']['header']['source'] == 'COLIN' \
+                and filing_json['filing']['header']['date'] < '2019-03-08':
+            return True
+
+        return False
+
     @staticmethod
     def _process_colin_filing(identifier: str, filing: Filing, business: Business) -> Tuple[dict, int]:
         try:
+
             if not filing.colin_event_ids:
                 raise KeyError
-            if not ListFilingResource._is_before_epoch_filing(filing.filing_json, business):
-                payload = {'filing': {'id': filing.id}}
-                queue.publish_json(payload)
-            else:
+
+            epoch_filing = Filing.get_filings_by_status(business_id=business.id, status=[Filing.Status.EPOCH.value])
+
+            if epoch_filing and ListFilingResource._is_before_epoch_filing(filing.filing_json, business):
                 epoch_filing = Filing.get_filings_by_status(business_id=business.id, status=[Filing.Status.EPOCH.value])
```

Comment:
Instead of creating this twice, can you include it with a walrus :=  in the if block?

Reply:
Good call.  Have updated as per your suggestion.